Example/HorizonCalendarExample/HorizonCalendarExample/Demo View Controllers/SwiftUIScreenDemoViewController.swift

```diff
@@ -136,17 +133,37 @@ struct SwiftUIScreenDemo: View {
     }
 
     .onDaySelection { day in
-      switch calendarSelection {
-      case .singleDay(let selectedDay):
-        if day > selectedDay {
-          calendarSelection = .dayRange(selectedDay...day)
-        } else {
-          calendarSelection = .singleDay(day)
-        }
-      case .none, .dayRange:
-        calendarSelection = .singleDay(day)
-      }
+      DayRangeSelectionHelper.updateDayRange(
+        afterTapSelectionOf: day,
+        existingDayRange: &selectedDayRange)
     }
+
+    .onMultipleDaySelectionDrag(
+      began: { day in
+        DayRangeSelectionHelper.updateDayRange(
+          afterDragSelectionOf: day,
+          existingDayRange: &selectedDayRange,
+          initialDayRange: &selectedDayRangeAtStartOfDrag,
+          state: .began,
+          calendar: calendar)
+      },
+      changed: { day in
+        DayRangeSelectionHelper.updateDayRange(
+          afterDragSelectionOf: day,
+          existingDayRange: &selectedDayRange,
+          initialDayRange: &selectedDayRangeAtStartOfDrag,
+          state: .changed,
+          calendar: calendar)
+      },
+      ended: { day in
+        DayRangeSelectionHelper.updateDayRange(
+          afterDragSelectionOf: day,
+          existingDayRange: &selectedDayRange,
+          initialDayRange: &selectedDayRangeAtStartOfDrag,
+          state: .ended,
+          calendar: calendar)
+      })
```

Comment:
alternatively, we could have a single closure that takes a gesture state param that we define (since we don't want to pass `UIGestureRecognizer.State` to SwiftUI code, feels dirty)